### PR TITLE
feature/IO-751_CI/CD

### PR DESCRIPTION
**Pull request by egor-slatvinskiy-idf**

### CODE REVIEW

**Review by vladimir-hodaryonok-idf — APPROVED** [review]

**Review by elizaveta-seleznyova-idf — APPROVED** [review]

**Review by alexei-kruk-idf — APPROVED** [review]

**Review by yuliya-klescheva-idf — APPROVED** [review]

File: .github/workflows/build_master.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  after_master_commit:
+    name: Build after master commit
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '11'
+      - uses: subosito/flutter-action@v1
```
Comment (sergei-mikhailovskii-idf) [inline .github/workflows/build_master.yml]: use latest version

File: .github/workflows/build_master.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        run: |
+          cd android;
+          touch "local.properties";
+          echo "${{secrets.FACEBOOK_APP_ID}}" >> local.properties;
+          echo "${{secrets.FB_LOGIN_PROTOCOL_SCHEME}}" >> local.properties;
+          echo "${{secrets.FACEBOOK_CLIENT_TOKEN}}" >> local.properties;
+      - name: Create api key
+        run: |
+          touch "prod.json";
+          touch "sandbox.json";
```
Comment (sergei-mikhailovskii-idf) [inline .github/workflows/build_master.yml]: why do u create sandbox.json? seems that we don't need it for this build

File: .github/workflows/build_master.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+          echo "${{secrets.PROD_API_KEY}}" >> prod.json;
+      - name: Pub get domain
+        run: |
+          cd packages/domain/
```
Comment (sergei-mikhailovskii-idf) [inline .github/workflows/build_master.yml]: use working_directory

File: .github/workflows/build_master.yml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+          flutter pub get && flutter packages pub run build_runner build
+      - name: Pub get data
+        run: |
+          cd packages/data/
```
Comment (sergei-mikhailovskii-idf) [inline .github/workflows/build_master.yml]: -//-

File: .github/workflows/build_master.yml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+          flutter pub get
+      - name: Pub get presentation
+        run: |
+          cd packages/presentation/
```
Comment (sergei-mikhailovskii-idf) [inline .github/workflows/build_master.yml]: -//-

File: .github/workflows/build_pr.yml [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+  pull_request_build:
+    name: Build after PR
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '11'
+      - uses: subosito/flutter-action@v1
```
Comment (sergei-mikhailovskii-idf) [inline .github/workflows/build_pr.yml]: -//-

File: .github/workflows/build_pr.yml [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+      - name: Create api key for android
+        run: |
+          cd android;
+          touch "local.properties";
+          echo "${{secrets.FACEBOOK_APP_ID}}" >> local.properties;
+          echo "${{secrets.FB_LOGIN_PROTOCOL_SCHEME}}" >> local.properties;
+          echo "${{secrets.FACEBOOK_CLIENT_TOKEN}}" >> local.properties;
+      - name: Create api key
+        run: |
+          touch "prod.json";
```
Comment (sergei-mikhailovskii-idf) [inline .github/workflows/build_pr.yml]: -//-

File: .github/workflows/build_pr.yml [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+          touch "sandbox.json";
+          echo "${{secrets.SANDBOX_API_KEY}}" >> sandbox.json;
+      - name: Pub get domain
+        run: |
+          cd packages/domain/
```
Comment (sergei-mikhailovskii-idf) [inline .github/workflows/build_pr.yml]: -//-

File: .github/workflows/build_pr.yml [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+          flutter pub get && flutter packages pub run build_runner build
+      - name: Pub get data
+        run: |
+          cd packages/data/
```
Comment (sergei-mikhailovskii-idf) [inline .github/workflows/build_pr.yml]: -//-

File: .github/workflows/build_pr.yml [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+          flutter pub get
+      - name: Pub get presentation
+        run: |
+          cd packages/presentation/
```
Comment (sergei-mikhailovskii-idf) [inline .github/workflows/build_pr.yml]: -//-

File: .github/workflows/build_pr.yml [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+          flutter pub global activate intl_utils 2.7.0
+          flutter pub get
+          flutter pub global run intl_utils:generate
+      - name: Pub get app
+        run: |
+          flutter pub get
+      - name: Build APK
+        run: flutter build apk --flavor sandbox -t lib/main_sandbox.dart --debug --dart-define=sandbox=true
+      - name: Firebase app distribution push
+        uses: wzieba/Firebase-Distribution-Github-Action@v1.3.5
```
Comment (sergei-mikhailovskii-idf) [inline .github/workflows/build_pr.yml]: -//-

File: packages/data/lib/database/database_repository_impl.dart [hunk 12]
```diff
@@ -62,7 +67,8 @@ class DatabaseRepositoryImpl implements DatabaseRepository {
     return await instanceDb.query(
       ConfigurationDatabase.movieList,
       where: '${ConfigurationDatabase.movieType} = '
-          '${type == TypeListMovie.trending
+          '${type
+          == TypeListMovie.trending
           ? ConfigurationDatabase.movieTrending
           : ConfigurationDatabase.movieComing}',
```
Comment (sergei-mikhailovskii-idf) [inline packages/data/lib/database/database_repository_impl.dart]: move it to separate variable

File: packages/data/lib/database/database_repository_impl.dart [hunk 13]
```diff
@@ -112,7 +118,8 @@ class DatabaseRepositoryImpl implements DatabaseRepository {
     await instanceDb.update(
       ConfigurationDatabase.dateLoad,
       where: '${ConfigurationDatabase.movieType} = '
-          '${typeMovie == TypeListMovie.trending
+          '${typeMovie
+          == TypeListMovie.trending
           ? ConfigurationDatabase.movieTrending
           : ConfigurationDatabase.movieComing}',
```
Comment (sergei-mikhailovskii-idf) [inline packages/data/lib/database/database_repository_impl.dart]: -//-

File: packages/data/lib/database/database_repository_impl.dart [hunk 14]
```diff
@@ -127,11 +134,13 @@ class DatabaseRepositoryImpl implements DatabaseRepository {
     final result = await instanceDb.query(
       ConfigurationDatabase.dateLoad,
       where: '${ConfigurationDatabase.movieType} = '
-          '${typeMovie == TypeListMovie.trending
+          '${typeMovie
+          == TypeListMovie.trending
           ? ConfigurationDatabase.movieTrending
           : ConfigurationDatabase.movieComing}',
```
Comment (sergei-mikhailovskii-idf) [inline packages/data/lib/database/database_repository_impl.dart]: -//-

**Review by sergei-mikhailovskii-idf — CHANGES_REQUESTED** [review]

File: .github/workflows/build_master.yml (hunk 2 above)
Comment (egor-slatvinskiy-idf) [reply .github/workflows/build_master.yml]: its use is written in the assets, without this option - an error

**Review by sergei-mikhailovskii-idf — APPROVED** [review]

**Pull request merged**